modAionImpl/src/org/aion/zero/impl/pendingState/AionPendingStateImpl.java

```diff
@@ -119,57 +118,79 @@ private synchronized void backupPendingTx() {
         pendingTxCache.clearCacheTxHash();
     }
 
-    public static AionPendingStateImpl create(CfgAion cfgAion, AionBlockchainImpl blockchain) {
-        AionPendingStateImpl ps = new AionPendingStateImpl(cfgAion);
-        ps.init(blockchain);
-        return ps;
+    public static AionPendingStateImpl create(
+            AionBlockchainImpl blockchain,
+            long energyUpperBound,
+            int txPendingTimeout,
+            int maxTxCacheSize,
+            boolean seedMode,
+            boolean poolBackup,
+            boolean poolDump) {
+
+        return new AionPendingStateImpl(
+                blockchain,
+                energyUpperBound,
+                txPendingTimeout,
+                maxTxCacheSize,
+                seedMode,
+                poolBackup,
+                poolDump);
     }
 
-    public static AionPendingStateImpl createForTest(CfgAion cfgAion, AionBlockchainImpl blockchain) {
-        AionPendingStateImpl ps = new AionPendingStateImpl(cfgAion);
-        ps.initForTest(blockchain);
+    public static AionPendingStateImpl createForTest(
+            AionBlockchainImpl blockchain,
+            long energyUpperBound,
+            int txPendingTimeout,
+            int maxTxCacheSize,
+            boolean seedMode,
+            boolean poolBackup,
+            boolean poolDump) {
+
+        AionPendingStateImpl ps =
+                new AionPendingStateImpl(
+                        blockchain,
+                        energyUpperBound,
+                        txPendingTimeout,
+                        maxTxCacheSize,
+                        seedMode,
+                        poolBackup,
+                        poolDump);
+        ps.testingMode = true;
+        ps.poolDumpEnable = !seedMode;
         return ps;
     }
 
-    private AionPendingStateImpl(CfgAion _cfgAion) {
-
-        this.isSeed = _cfgAion.getTx().isSeedMode();
+    private AionPendingStateImpl(
```

Comment:
Could we make this constructor public and avoid the need for the two static factory methods?